apps/sms/views/shared/js/navigation.js

```diff
@@ -1,361 +1,726 @@
+  *   hash if the same HTML file holds several views.
+  * - url: this is the URL used to match the current location and find the
+  *   current view.
+  * - view: this is the actual JavaScript object that manages this view.
+  * - previous (optional): this is the name of the previous view; this is
+  *   necessary to decide if we do a forward/backward animation.
+  * - partOf (optional): this is used when a view is part of another view's
+  *   HTML document. In that case one view is "master" and the other are only
+  *   part of the master view.
+  * - alsoContains (optional): this is the opposite and holds the view's
+  *   subviews. (currently unused)
+  *
+  * NGA_VIEWS holds the configuration for NGA split views.
+  */
+const NGA_VIEWS = Object.freeze([
+  Object.freeze({
+    name: 'thread',
+    url: 'views/conversation/index.html',
+    view: 'ConversationView',
+    previous: 'thread-list',
+    alsoContains: ['report-view', 'composer', 'group-view']
+  }),
+  Object.freeze({
+    name: 'thread-list',
+    url: 'views/inbox/index.html',
+    view: 'InboxView'
+  }),
+  Object.freeze({
+    name: 'report-view',
+    view: 'ReportView',
+    previous: 'thread',
+    partOf: 'thread'
+  }),
+  Object.freeze({
+    name: 'composer',
+    view: 'ConversationView',
+    previous: 'thread-list',
+    partOf: 'thread'
+  }),
+  Object.freeze({
+    name: 'group-view',
+    view: 'GroupView',
+    previous: 'thread',
+    partOf: 'thread'
+  })
+]);
+
+/**
+  * OA_VIEWS holds the configuration for the old non-split architecture.
+  */
+const OA_VIEWS = Object.freeze([
+  Object.freeze({
+    name: 'thread-list',
+    url: 'index.html',
+    view: 'InboxView',
+    alsoContains: ['thread', 'composer', 'group-view', 'report-view']
+  }),
+  Object.freeze({
+    name: 'thread',
+    view: 'ConversationView',
+    previous: 'thread-list',
+    partOf: 'thread-list'
+  }),
+  Object.freeze({
+    name: 'composer',
+    view: 'ConversationView',
+    previous: 'thread-list',
+    partOf: 'thread-list'
+  }),
+  Object.freeze({
+    name: 'group-view',
+    view: 'GroupView',
+    previous: 'thread',
+    partOf: 'thread-list'
+  }),
+  Object.freeze({
+    name: 'report-view',
+    view: 'ReportView',
+    previous: 'thread',
+    partOf: 'thread-list'
+  })
+]);
+
+const VIEWS = isUsingOldArchitecture() ? OA_VIEWS : NGA_VIEWS;
+
+var currentState;
+var navigationTransition;
+
+/**
+ * Navigate to this URL.
+ *
+ * @param {String} url URL to load.
+ */
+function setLocation(url) {
+  window.location.assign(url);
+}
+
+/**
+ * We're using the old architecture if we're not in a split document.
+ *
+ * @returns {Boolean} true if we're using the old architecture.
+ */
+function isUsingOldArchitecture() {
+  var pathname = window.location.pathname;
+  return (pathname === '/' || pathname === '/index.html');
+}
+
+/**
+ * Tries to find a view name from a URL hash.
+ * It takes the part of the hash between '/' or the start of the string, and '?'
```

Comment:
nit: could you please also mention few examples of hashes we support here, so that it's easier to understand what string we expect here?